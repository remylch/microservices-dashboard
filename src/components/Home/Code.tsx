import React from "react";
import { IoGitCommit } from "react-icons/io5";
import { fetchGitlabProjects, fetchProjectCommits } from "../../api/gitlabAPI";
import TopBar from "../Settings/TopBar";

const Code = () => {
  const [idProjectSelected, setIdProjectSelected] = React.useState<
    number | undefined
  >();

  const [commits, setCommits] = React.useState([]);
  const [projects, setProjects] = React.useState([]);
  React.useEffect(() => {
    fetchGitlabProjects().then((res: any) => {
      setProjects(res.data);
    });
  }, []);

  React.useEffect(() => {
    if (idProjectSelected)
      fetchProjectCommits(idProjectSelected).then((res) => {
        console.log(res);
        setCommits(res.data);
      });
    return () => {
      setCommits([]);
    };
  }, [idProjectSelected]);

  return (
    <div className="h-full w-full flex flex-1 flex-col min-h-screen">
      <TopBar projects={projects} setIdProjectSelected={setIdProjectSelected} />
      <div className="p-10 flex h-full w-full flex-col gap-5">
        {idProjectSelected ? (
          <>
            <h3 className="text-gray-400 ">Latests commits</h3>
            {commits.map((commit: any) => {
              const date = commit.created_at.split("T")[0];
              const hour = date.split("+")[0];

              return (
                <div className="flex flex-col px-5 py-2" key={commit.id}>
                  <div className="font-ligh text-sm text-gray-500 flex gap-3">
                    <label>{date}</label>
                    <label>{hour}</label>
                  </div>
                  <div className="flex items-center gap-2">
                    <span className="text-black">{commit.author_name}</span>
                    <IoGitCommit className="w-5 h-5 pt-1" />
                    <q className="text-gray-600 text-sm font-semibold">
                      {commit.message}
                    </q>
                  </div>
                </div>
              );
            })}
          </>
        ) : (
          <div className="flex font-semibold flex-1 justify-center items-center text-gray-500">
            <h1>No project selected.</h1>
          </div>
        )}
      </div>
    </div>
  );
};

export default Code;
