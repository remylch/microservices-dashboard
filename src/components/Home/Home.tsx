import React from "react"

const Home: React.FC = () => <div></div>

export default Home
