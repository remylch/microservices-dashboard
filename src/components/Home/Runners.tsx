import React from "react"
import Runner from "../Runners/runner"
import { RunnerType } from "../../types/runner-types"
import useRunners from "../../hooks/runners-hooks/useRunners"

const Runners = () => {
  const { runners, runningsRunners } = useRunners()

  return (
    <div className="flex flex-col p-5 w-full">
      <div className="relative border border-gray-300 mb-10">
        <h3 className="absolute -top-3 left-8 bg-white px-3">Running</h3>
      </div>
      <div className="flex gap-5 items-start content-start flex-wrap ">
        {runningsRunners.length > 0 ? (
          runningsRunners.map((runner: RunnerType) => (
            <Runner
              name={runner.name}
              id={runner.id}
              active={runner.active}
              description={runner.description}
              ipAddress={runner.ipAddress}
              isShared={runner.isShared}
              online={runner.online}
              paused={runner.paused}
              runnerType={runner.runnerType}
              status={runner.status}
              key={runner.id}
            />
          ))
        ) : (
          <label data-testid="message-no-runner-running">
            No runner is currently running.
          </label>
        )}
      </div>
      <div className="relative border border-gray-300 my-10">
        <h3 className="absolute -top-3 left-8 bg-white px-3">Terminated</h3>
      </div>
      {runners.length > 0 ? (
        runners.map((runner: RunnerType) => (
          <Runner
            name={runner.name}
            id={runner.id}
            active={runner.active}
            description={runner.description}
            ipAddress={runner.ipAddress}
            isShared={runner.isShared}
            online={runner.online}
            paused={runner.paused}
            runnerType={runner.runnerType}
            status={runner.status}
            key={runner.id}
          />
        ))
      ) : (
        <label data-testid="message-no-runner-available">
          No runner currently available.
        </label>
      )}
    </div>
  )
}

export default Runners
