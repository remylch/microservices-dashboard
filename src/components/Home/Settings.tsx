import InputText from "../utils-components/Inputs"
import React, { useCallback, useEffect, useState } from "react"
import { clickOnButtonFieldDisabled } from "../../utils/toastUtils"

type SettingsFields = {
  gitlabDomain: string
  gitlabToken: string
  servicesDomain: string
  servicesToken: string
}

const Settings = () => {
  const [fields, setFields] = useState<SettingsFields>({
    gitlabDomain: "",
    gitlabToken: "",
    servicesDomain: "",
    servicesToken: "",
  })
  const [originalFields, setOriginalFields] = useState<SettingsFields>({
    gitlabDomain: "",
    gitlabToken: "",
    servicesDomain: "",
    servicesToken: "",
  })

  useEffect(() => {
    //PATCH: setup originals fields by getting api call
    return () => {
      setOriginalFields({
        gitlabDomain: "",
        gitlabToken: "",
        servicesDomain: "",
        servicesToken: "",
      })
    }
  }, [])

  const updateValue = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
    const { name, value } = e.target
    setFields({ ...fields, [name]: value })
  }, [])

  function checkFieldUpdate(field: string): boolean {
    const myfield = field as keyof typeof fields
    return fields[myfield] !== originalFields[myfield] ? true : false
  }

  return (
    <div className="h-full  flex flex-1 flex-col min-h-screen relative z-10">
      <section className="flex flex-1 my-10 mx-8 ">
        <div className="flex flex-col mt-5 w-full">
          {/* gitlab informations */}
          <div className="shadow-md border border-gray-200 w-full p-5 pt-7 rounded-lg mb-10 relative">
            <span className="absolute z-10 left-5 -top-4 bg-white dark:bg-fade-blue px-3 text-gray-500 dark:text-secondary">
              Gitlab Informations
            </span>
            <h2 className="mb-5 font-semibold text-primary dark:text-secondary">
              Gitlab DNS
            </h2>
            <div className="flex mb-5 gap-5">
              <InputText
                name="gitlabDomain"
                value={fields.gitlabDomain}
                setValue={updateValue}
              />
              {checkFieldUpdate("gitlabDomain") ? (
                <button className="btn-primary">Save</button>
              ) : (
                <button
                  className="btn-disabled"
                  onClick={() => clickOnButtonFieldDisabled()}
                >
                  Save
                </button>
              )}
            </div>
            <h2 className="mb-3 font-semibold text-primary dark:text-secondary">
              Gitlab Token
            </h2>
            <div className="flex gap-5">
              <InputText
                name="gitlabToken"
                value={fields.gitlabToken}
                setValue={updateValue}
              />
              {checkFieldUpdate("gitlabToken") ? (
                <button className="btn-primary">Save</button>
              ) : (
                <button
                  className="btn-disabled"
                  onClick={() => clickOnButtonFieldDisabled()}
                >
                  Save
                </button>
              )}
            </div>
          </div>
          {/* services informations */}
          <div className="shadow-md border border-gray-200 w-full p-5 pt-7 rounded-lg mb-10 relative">
            <span className="absolute z-10 left-5 -top-4 bg-white px-3 dark:bg-fade-blue text-gray-500 dark:text-secondary">
              Services Informations
            </span>
            <h2 className="mb-5 font-semibold text-primary dark:text-secondary">
              Services up :
            </h2>
            <h2 className="mb-5 font-semibold text-primary dark:text-secondary">
              Services Down :
            </h2>
            <h2 className="mb-5 font-semibold text-primary dark:text-secondary">
              Services Error :
            </h2>
            <h2 className="mb-5 font-semibold text-primary dark:text-secondary">
              Services URL
            </h2>
            <div className="flex mb-5 gap-5">
              <InputText
                name="servicesDomain"
                setValue={updateValue}
                value={fields.servicesDomain}
              />
              {checkFieldUpdate("servicesDomain") ? (
                <button className="btn-primary">Save</button>
              ) : (
                <button
                  className="btn-disabled"
                  onClick={() => clickOnButtonFieldDisabled()}
                >
                  Save
                </button>
              )}
            </div>
            <h2 className="mb-3 font-semibold text-primary dark:text-secondary">
              Access token
            </h2>
            <div className="flex gap-5">
              <InputText
                name="servicesToken"
                setValue={updateValue}
                value={fields.servicesToken}
              />
              {checkFieldUpdate("servicesToken") ? (
                <button className="btn-primary">Save</button>
              ) : (
                <button
                  className="btn-disabled"
                  onClick={() => clickOnButtonFieldDisabled()}
                >
                  Save
                </button>
              )}
            </div>
          </div>
          {/* bottom buttons */}
          <div className="shadow-md flex flex-col border border-gray-200 w-full p-5 pt-7 rounded-lg mb-5 relative">
            <span className="absolute z-10 left-5 -top-4 bg-white px-3 dark:bg-fade-blue  text-gray-500 dark:text-secondary">
              Account
            </span>
            <div className="w-full flex flex-col">
              <div className="flex items-center gap-5 py-5">
                <span className="font-semibold   text-primary dark:text-secondary">
                  Email
                </span>
                <span className="text-primary dark:text-secondary">
                  test@mail.com
                </span>
                <button className="text-sm font-light text-helio-gray hover:underline cursor-pointer hover:underline-offset-2">
                  modify
                </button>
              </div>

              <div className="border-b border-gray-300" />
              <div className="flex items-center gap-5 py-5">
                <span className="font-semibold text-primary dark:text-secondary">
                  Password
                </span>
                <span className="text-primary dark:text-secondary">
                  ************
                </span>
                <button className="text-sm font-light  text-helio-gray hover:underline cursor-pointer hover:underline-offset-2">
                  modify
                </button>
              </div>
            </div>

            <div className="border-b border-gray-300 mb-5" />
            <button className="btn-danger self-center">Delete account</button>
          </div>
        </div>
      </section>
    </div>
  )
}

export default Settings
