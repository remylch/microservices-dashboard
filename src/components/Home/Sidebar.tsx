import React from "react"
import { AiFillCode, AiFillSetting } from "react-icons/ai"
import {
  FaRunning,
  FaChevronRight,
  FaChevronLeft,
  FaDocker,
} from "react-icons/fa"
import { Link, useLocation } from "react-router-dom"
import { IoMdExit } from "react-icons/io"
import { SizeContext } from "../../hooks/size-observer"
import gitlabLogo from "../../statics/images/gitlab-logo-500.svg"
import { MdOutlineApps } from "react-icons/md"

const links = [
  {
    name: "code",
    Icon: AiFillCode,
  },
  {
    name: "services",
    Icon: MdOutlineApps,
  },
  {
    name: "containers",
    Icon: FaDocker,
  },
  {
    name: "runners",
    Icon: FaRunning,
  },
  {
    name: "settings",
    Icon: AiFillSetting,
  },
]

const Sidebar = () => {
  const [isOpen, setIsOpen] = React.useState<boolean>(true)

  const location = useLocation()

  const { innerWidth } = React.useContext(SizeContext)

  React.useEffect(() => {
    if (innerWidth <= 800) {
      setIsOpen(false)
    } else {
      setIsOpen(true)
    }
  }, [innerWidth])

  const toggleSidebar = () => {
    setIsOpen((open) => !open)
  }

  return (
    <div
      className={`max-h-screen transition sticky left-0 top-0 transform-cpu duration-300 ease-in items-stretch justify-between bg-blue-500 min-h-full ${
        isOpen ? "w-1/4 min-w-min max-w-min" : "w-fit"
      } py-10`}
    >
      <div>
        <div
          role="button"
          tabIndex={0}
          onClick={toggleSidebar}
          onKeyDown={toggleSidebar}
          className={`text-gray-200 flex self-end cursor-pointer ${
            isOpen ? "justify-end" : "justify-center"
          } px-5 w-full mb-10`}
        >
          {isOpen ? <FaChevronLeft /> : <FaChevronRight />}
        </div>
        <div
          className={`font-semibold text-xl flex items-center  justify-center pb-10 text-white ${
            isOpen && "pr-10"
          }`}
        >
          <img alt="logo gitlab" className="h-16" src={gitlabLogo} />
          {isOpen && <h2>Gitlab Dashboard</h2>}
        </div>

        {links.map((link) => (
          <Link
            to={link.name}
            key={link.name}
            className={`group ${
              link.name === links[links.length - 1].name && "mb-5"
            } ${
              location.pathname === `/${link.name}` && "bg-blue-700"
            } cursor-pointer w-full h-14 capitalize flex items-center py-2 relative ${
              isOpen ? "px-10" : "px-2 justify-center"
            } text-white hover:bg-blue-600`}
          >
            <link.Icon className={`h-10 ${isOpen && "mr-5"}`} />

            {isOpen ? (
              link.name
            ) : (
              <label
                id="labelLinkName"
                className="absolute z-[9999] bg-white hidden group-hover:block px-3 py-1 rounded-full border border-gray-300 shadow-sm top-1/2 translate-y-[-50%] left-[110%] duration-300 ease-in-out transform text-blue-500"
              >
                {link.name}
              </label>
            )}
            {isOpen && location.pathname === `/${link.name}` && (
              <FaChevronRight className="mt-1 absolute right-5" />
            )}
          </Link>
        ))}
      </div>

      <button
        className={`bg-blue-700 mx-5 ${
          isOpen ? "px-10 py-3" : " w-fit p-3"
        }   rounded-md text-white hover:bg-blue-800 cursor-pointer`}
      >
        {isOpen ? <span>Disconnect</span> : <IoMdExit />}
      </button>
    </div>
  )
}

export default Sidebar
