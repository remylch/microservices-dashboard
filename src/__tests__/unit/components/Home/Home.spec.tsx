/**
 * @jest-environment jsdom
 */

import renderer from "react-test-renderer"
import Home from "../../../../components/Home/Home"

describe("Home component tests", () => {
  it("should render the component correctly", () => {
    const home = renderer.create(<Home />)
    expect(home.toJSON()).toMatchSnapshot()
  })
})

export default {}
