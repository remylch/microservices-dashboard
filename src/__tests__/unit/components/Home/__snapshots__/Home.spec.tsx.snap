// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component tests should render the component correctly 1`] = `<div />`;
